packages/codeeditor/src/viewer.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { StackedLayout, Widget } from '@lumino/widgets';
+import { CodeEditor, CodeEditorWrapper } from './';
+
+export class CodeViewerWidget extends Widget {
+  /**
+   * Construct a new code viewer widget.
+   */
+  constructor(options: CodeViewerWidget.IOptions) {
+    super();
+
+    this.model = new CodeEditor.Model({
+      value: options.content,
+      mimeType: options.mimeType
+    });
+
+    const editorWidget = new CodeEditorWrapper({
+      factory: options.factory,
+      model: this.model
+    });
+    this.editor = editorWidget.editor;
+    this.editor.setOption('readOnly', true);
+
+    const layout = (this.layout = new StackedLayout());
+    layout.addWidget(editorWidget);
+  }
+
+  public model: CodeEditor.IModel;
+  public editor: CodeEditor.IEditor;
```

Comment:
The rules is usually to drop the `public` visibility. Should they be readonly?

```suggestion
  model: CodeEditor.IModel;
  editor: CodeEditor.IEditor;
```

Reply:
yeah I can change that, I copied this format from the FileEditor and did think it was odd